lib/irb/input-method.rb

```diff
@@ -175,10 +175,15 @@ def close
   class ReadlineInputMethod < StdioInputMethod
     class << self
       def initialize_readline
-        require "readline"
-      rescue LoadError
-      else
-        include ::Readline
+        return if defined?(self::Readline)
+
+        begin
+          require 'readline'
+          const_set(:Readline, ::Readline)
+        rescue LoadError
+          const_set(:Readline, ::Reline)
+        end
```

Comment:
Adds `ReadlineInputMethod::Readline = ::Readline or ::Reline`
because we don't want to pollute the top level by adding constant `::Readline` when `require 'readline'` fails.